lib/core/src/lib.rs

```diff
@@ -5,6 +5,7 @@
 #![feature(specialization)]
 #![feature(associated_type_defaults)]
 #![feature(set_stdio)]
+#![feature(arbitrary_self_types)]
```

Comment:
Why do we need to use this feature? I mean, we already use too many unstable features and I dont see a reason for this one. Of course there may be a legimitate use case - i'd love to chat about it! :)

Reply:
I used it in the `EventHandler` to help me write less code. Please check the implementation and usage of `EventHandler` in `Navigator` struct.

Reply:
Is it still valid? I don't see `EventHandler` anymore.

Reply:
@wdanilo nope. It was refactored to `MouseManager`

Reply:
I don't see usages of `arbitrary_self_types` there. Would you be so nice and point me to where this extension is used?

Reply:
It is no longer needed. I removed it.